src/System.IO.Pipelines.Networking.Libuv/UvTcpConnection.cs

```diff
@@ -214,6 +223,8 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
                         }
                     }
                 }
+
+                inputBufferHandle.Free();
```

Comment:
Hmm sorry, do you mind explaining why that's the case? If we commit but keep the buffer then wouldn't it be a problem to drop the handle if it's still going to be used?